pkg/controller/infrastructure/actuator.go

```diff
@@ -561,3 +565,24 @@ func (a *actuator) Migrate(ctx context.Context, infra *extensionsv1alpha1.Infras
 
 	return tf.CleanupConfiguration(ctx)
 }
+
+// checkAndCreateServiceLinkedRole is to check if service linked role exists, if not create one.
+func (a *actuator) checkAndCreateServiceLinkedRole(ctx context.Context, infra *extensionsv1alpha1.Infrastructure, credentials *alicloud.Credentials) error {
```

Comment:
I would suggest to change `checkAndCreateServiceLinkedRole` to `ensureServiceLinkedRole`